Cargo.toml

```diff
@@ -12,6 +12,7 @@ members = [
 
 [workspace.dependencies]
 # Third party dependencies
+arrayvec = { version="0.7.4", default-features=false }
```

Comment:
Is there any reason to use arrayvec over tinyvec or smallvec?

My main preference goes to tinyvec since it is free from unsafe code, the only compromise is that we would need to restructure thing such that the type of data we store should implement default.

Reply:
> Is there any reason to use arrayvec over tinyvec or smallvec?

Only personal familiarity :) I'm open to using something else, but implementing default seems a bit awkward: how do you envision doing so? Ideally we would avoid a present-but-unset state that doesn't mean anything other than (probably) a bug somewhere, and I think we do want to keep Session as a trait to allow arbitrary impls, although this might not be essential.

Reply:
Got it, thanks for the discussion! I'm going to leave it as-is for now, since it doesn't seem like a clear improvement to switch, but like you I don't have a preference either way so no objections to changing it in the future if doing so makes sense.